shell/plugins/rancher-api/cluster-api-class.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * ClusterApi extends the BaseClusterApi to provide specific functionalities for interacting with
+ * resources within the current cluster context. It includes methods to create, edit, and delete
+ * resources. This class is specifically designed for the $cluster context and should not be
+ * confused with CAPI.
+ */
+
+import { SteveResource, ResourceManageOptions } from '@shell/types/rancher-api/cluster';
+import BaseClusterApi from './base-cluster-class';
+
+export default class ClusterApi extends BaseClusterApi {
```

Comment:
I have a suspicion we'll want to expose more of the methods from shell/store.index.js.

A few examples:
getCluster
getClusterId
isRancher

Reply:
I decided not to dive too into the weeds with implementation before we agreed on a proper plan of this API.